app/models/variant_override_set.rb

```diff
@@ -1,6 +1,8 @@
 class VariantOverrideSet < ModelSet
   def initialize(collection, attributes={})
     super(VariantOverride, collection, attributes, nil,
-          proc { |attrs| attrs['price'].blank? && attrs['count_on_hand'].blank? } )
+          # I have no idea what this does but had to add all fields to get it to create a new VO when only a new field was changed.
+          # i.e. a field unique to Variant Override (default_stock and reset) and not present in Variant.
+          proc { |attrs| attrs['price'].blank? && attrs['count_on_hand'].blank? && attrs['default_stock'].blank? && attrs['enable_reset'].blank?} )
```

Comment:
This proc loops over the VO that it has been handed in the method call and evaluates if all of the 'attrs' (price etc) are blank, or if any of them is not.

The super call refers to the class extended: ModelSet. Have a look at app/models/modelset.rb to see that essentially if this evaluates to true the VO is deleted. 

So that makes sense. If you don't add the additional fields, then only change one of these fields, the VO will be deleted.